src/convenience/webhook.ts

```diff
@@ -119,21 +66,25 @@ const frameworkAdapters: Record<SupportedFrameworks, FrameworkAdapter> = {
  *
  * @param bot The bot for which to create a callback
  * @param framework An optional string identifying the framework (default: 'express')
- * @param onTimeout An optional strategy to handle timeouts (default: 'throw')
- * @param timeoutMilliseconds An optional number of timeout milliseconds (default: 10_000)
+ * @param adapterOptions Additional options for adapter behavior
+ * @param adapterOptions.onTimeout An optional strategy to handle timeouts (default: 'throw')
+ * @param adapterOptions.timeoutMilliseconds An optional number of timeout milliseconds (default: 10_000)
  */
-export function webhookCallback<C extends Context = Context>(
+export function webhookCallback<
+    A extends FrameworkAdapter,
+    C extends Context = Context,
+>(
     bot: Bot<C>,
-    framework: SupportedFrameworks = "express",
-    onTimeout: "throw" | "return" | ((...args: any[]) => unknown) = "throw",
-    timeoutMilliseconds = 10_000,
-) {
-    const server = frameworkAdapters[framework] ?? standard;
+    adapter?: A,
```

Comment:
I missed this. This is a breaking change so we either have to wait with merging until 2.0 (some time later next year maybe?), or it has to be reverted.

Reply:
Alternatively, we could introduce a union type. This would still permit string identifiers (non-breaking) but optionally allow people to pass their own adapter. May make it hard to find good types, not sure.

Reply:
Sure, this should be in the next major. Thought I feel that giving a choice among predefined frameworks has less flexibility than allowing to put either a custom one or import from grammY's code. I'll add a temporary union

Reply:
I don't see why users would have to access the predefined objects. How about not exposing all of them, and simply allowing string identifiers as well as custom objects? One could pass a string to use a predefined one, or an object to define a custom one.

Reply:
Is there a reason for exposing them?

Reply:
A parameter like `adapter: BuiltInAdapters | AdapterObject` is ideal in my opinion.

Reply:
![image](https://user-images.githubusercontent.com/49933115/142662636-4030cb66-cc09-44d0-a863-42d321621fb2.png)
This way it's ideal

Reply:
That's also what I have in mind. Plus, letting the adapters become implementation details that aren't exported.